File: code-orig/cpp/note.cpp
#include "note.hpp"

Note::Note() {
  title = "";
  content = "";
}

Note::Note(const std::string &title) {
  this->title = title;
  content = "";
}

const std::string &Note::getTitle() const { return title; }

const std::string &Note::getContent() const { return content; }

void Note::appendContent(const std::string &text) { content += text; }

File: code-orig/cpp/rust/note.rs
pub struct Note {
    pub title: String,
    pub content: String,
}

impl Note {
    pub fn new(title: &str) -> Note {
        Note {
            title: title.to_string(),
            content: String::new(),
        }
    }

    pub fn append_content(&mut self, text: &str) {
        self.content += text;
    }

    pub fn get_content(&self) -> &str {
        &self.content
    }

    pub fn display(&self) {
        println!("Title: {}", self.title);
        println!("{}", self.content);
    }
}
